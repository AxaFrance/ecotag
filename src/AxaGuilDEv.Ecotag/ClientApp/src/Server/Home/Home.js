import React from "react";
import Title from "../../TitleBar";
import {Link} from 'react-router-dom';
import './Home.scss';
import withAuthentication, {Administateur, Annotateur, DataScientist} from "../withAuthentication";
import {OidcUserStatus} from "@axa-fr/react-oidc";

export const Home = ({user: {roles = []}, userLoadingState}) => (
    <div className="home">
        <Title title="Accueil" goButton={false}/>
        <div className="home__links-container">
            {roles.includes(Annotateur) && <Link className="home__link" to="/projects">
                <div className="home__link-container home__link-container--projects">Projets</div>
            </Link>}
            {roles.includes(DataScientist) && <Link className="home__link" to="/datasets">
                <div className="home__link-container home__link-container--datasets">Datasets</div>
            </Link>}
            {roles.includes(Administateur) && <Link className="home__link" to="/teams">
                <div className="home__link-container home__link-container--groups">Equipes</div>
            </Link>}
            {(!roles.includes(Annotateur) && !roles.includes(DataScientist) && !roles.includes(Administateur) && userLoadingState === OidcUserStatus.Loaded) &&
                <p>Vous n'avez aucun rôle attribué à votre profile.</p>}
        </div>
    </div>
);

export default withAuthentication()(Home);
