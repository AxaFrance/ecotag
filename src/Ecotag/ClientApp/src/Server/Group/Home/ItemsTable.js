import React, {useState} from "react";
import Table, {Paging} from '@axa-fr/react-toolkit-table';
import Action from '@axa-fr/react-toolkit-action';
import Edit from "./Edit";
import './ItemsTable.scss';
import useProjectTranslation from "../../../useProjectTranslation";

const computeUsersList = users => {
    if (users.length === 0) {
        return "-";
    }
    return users.map(user => user.email).join(', ');
};

const UserRow = ({id, name, users, eligibleUsers, onUpdateUser}) => {
    const [isManageUsersModalVisible, setManageUsersModalVisible] = useState(false);
    const {translate} = useProjectTranslation();
    return (
        <Table.Tr key={id}>
            <Table.Td>{name}</Table.Td>
            <Table.Td>{computeUsersList(users)}</Table.Td>
            <Table.Td class="af-table__cell af-table__small">
                <Action id="editActionId" icon="pencil" title={translate('group.list.edit')}
                        onClick={() => setManageUsersModalVisible(true)}/>
                <Edit
                    idGroup={id}
                    users={users}
                    eligibleUsers={eligibleUsers}
                    onUpdateUser={onUpdateUser}
                    isManageUsersModalVisible={isManageUsersModalVisible}
                    setManageUsersModalVisible={setManageUsersModalVisible}
                />
            </Table.Td>
        </Table.Tr>
    );
};

const ItemsTable = ({items, filters, onChangePaging, onUpdateUser}) => {
    const {translate} = useProjectTranslation();

    return (
        <>
            <Table>
                <Table.Header>
                    <Table.Tr>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('group.list.header.name')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('group.list.header.users')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('group.list.header.actions')}</span>
                        </Table.Th>
                    </Table.Tr>
                </Table.Header>
                <Table.Body>
                    {items.map(({id, name, users = [], eligibleUsers = []}) => (
                        <UserRow
                            key={id}
                            id={id}
                            name={name}
                            users={users}
                            eligibleUsers={eligibleUsers}
                            onUpdateUser={onUpdateUser}
                        />
                    ))}
                </Table.Body>
            </Table>
            <Paging
                onChange={onChangePaging}
                numberItems={filters.paging.numberItemsByPage}
                numberPages={filters.paging.numberPages}
                currentPage={filters.paging.currentPage}
                id="home_paging"
                displayLabel={translate('group.paging.display')}
                elementsLabel={translate('group.paging.elements')}
                previousLabel={translate('group.paging.previous')}
                nextLabel={translate('group.paging.next')}
            />
        </>
    )
};

export default ItemsTable;