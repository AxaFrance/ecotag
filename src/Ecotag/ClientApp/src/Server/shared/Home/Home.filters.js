export const filterPaging = (items, numberItems, currentPage) => {
    let pageIndex = 0;
    if (currentPage > 1) {
        pageIndex = currentPage - 1;
    }

    let controlledNumberItems = 0;
    if (numberItems > 0) {
        controlledNumberItems = numberItems;
    }

    const length = items.length;
    const beginIndex = pageIndex * controlledNumberItems;
    const lastIndex = length < beginIndex + controlledNumberItems ? length : beginIndex + controlledNumberItems;

    return items.slice(beginIndex, lastIndex);
};

export const computeNumberPages = (items, numberItems) => {
    if (!items) {
        return 1;
    }
    const numberItemsControlled = numberItems < 1 ? 1 : numberItems;
    const length = items.length;
    return length > 0 ? Math.ceil(length / numberItemsControlled) : 1;
};

export const getItemsFiltered = (items, filterValue) => {
    if (filterValue !== null && filterValue !== undefined && filterValue.length > 0) {
        return items.filter(project => project.name.toLowerCase().includes(filterValue.toLowerCase()));
    }
    return items;
};

export const sortByTimeLastUpdate = (a, b) => {
    if (a.value.timeLastUpdate === b.value.timeLastUpdate) {
        return 0;
    }

    if (a.value.timeLastUpdate === null) {
        return 1;
    }

    if (b.value.timeLastUpdate === null) {
        return -1;
    }

    return a.value.timeLastUpdate - b.value.timeLastUpdate;
};

const computeEntries = columns => {
    const entries = [];
    for (const [key, value] of Object.entries(columns)) {
        entries.push({key, value});
    }
    return entries;
};

const compareItems = (entry, itemA, itemB) => {
    if (typeof itemB[entry.key] === 'number') {
        return itemB[entry.key] < itemA[entry.key] ? 1 : -1;
    }
    if (itemB[entry.key] && itemA[entry.key]) {
        if (itemB[entry.key] instanceof Date) {
            return itemA[entry.key].getTime() - itemB[entry.key].getTime();
        }
        const localComp = itemA[entry.key].localeCompare(itemB[entry.key]);
        if (localComp !== 0) {
            return localComp;
        }
    }
};

const compareItemsBySortingType = (columns, entry, itemA, itemB) => {
    if (columns[entry.key].value === 'asc') {
        return compareItems(entry, itemA, itemB);
    }
    if (columns[entry.key].value === 'desc') {
        return compareItems(entry, itemB, itemA);
    }
};

export const getItemsSorted = (items, columns) => {
    const entries = computeEntries(columns);
    entries.sort(sortByTimeLastUpdate);
    const criteria = entries[0];
    return [...items].sort((itemA, itemB) => compareItemsBySortingType(columns, criteria, itemA, itemB));
};
