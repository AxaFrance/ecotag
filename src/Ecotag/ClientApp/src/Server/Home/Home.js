import React from "react";
import Title from "../../TitleBar";
import {Link} from 'react-router-dom';
import './Home.scss';
import withAuthentication, {Administateur, Annotateur, DataScientist} from "../withAuthentication";
import {OidcUserStatus} from "@axa-fr/react-oidc";
import {useTranslation} from "react-i18next";

export const Home = ({user: {roles = []}, userLoadingState}) => {
    const {t} = useTranslation();

    return (
        <div className="home">
            <Title title={t('home.title')} goButton={false}/>
            <div className="home__links-container">
                {roles.includes(Annotateur) && <Link className="home__link" to="/projects">
                    <div className="home__link-container home__link-container--projects">{t('home.projects')}</div>
                </Link>}
                {roles.includes(DataScientist) && <Link className="home__link" to="/datasets">
                    <div className="home__link-container home__link-container--datasets">{t('home.datasets')}</div>
                </Link>}
                {roles.includes(Administateur) && <Link className="home__link" to="/teams">
                    <div className="home__link-container home__link-container--groups">{t('home.groups')}</div>
                </Link>}
                {(!roles.includes(Annotateur) && !roles.includes(DataScientist) && !roles.includes(Administateur) && userLoadingState === OidcUserStatus.Loaded) &&
                    <p>{t('home.errors.no_roles')}</p>}
            </div>
        </div>
    );
}

export default withAuthentication()(Home);
