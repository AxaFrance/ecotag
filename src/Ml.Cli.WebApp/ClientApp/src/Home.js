import React from "react";
import {Header, Name} from "@axa-fr/react-toolkit-layout-header";
import {Link} from 'react-router-dom';
import logo from '@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg';
import './Home.scss';
import Title from "@axa-fr/react-toolkit-layout-header/dist/Title/Title";

const Home = () => (
    <div className="home">
        <Header>
            <Name
                title="ML-CLI"
                subtitle="Made by AXA"
                img={logo}
                alt="Logo AXA"
            />
        </Header>
        <Title title="Liste des services"/>
        <div className="home__links-container">
            <Link className="home__link" to="/compare">
                <div className="home__link-container home__link-container--compare">Comparaison de services</div>
            </Link>
            <Link className="home__link" to="/annotate">
                <div className="home__link-container home__link-container--annotate">Annotation</div>
            </Link>
        </div>
    </div>
);

export default Home;
