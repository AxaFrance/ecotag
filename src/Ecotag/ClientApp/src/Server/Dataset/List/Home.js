import React from 'react';
import {Link} from 'react-router-dom';
import Title from '../../../TitleBar';
import './Home.scss';
import ItemsTable from "./ItemsTable";
import EmptyArrayManager from "../../../EmptyArrayManager";
import useProjectTranslation from "../../../useProjectTranslation";

const Home = ({items, numberTotalItems, filters, onChangePaging, onChangeFilter}) => {
    const {translate} = useProjectTranslation();
    return (
        <>
            <Title title={translate('dataset.title')} subtitle={translate('dataset.subtitle')}/>
            <div className="af-home container">
                <Link className="btn af-btn af-btn--quote" to="/datasets/new">
                    <span className="af-btn__text">{translate('dataset.new_dataset')}</span>
                </Link>
                <a className="af-home__access-token" target="_self" href="/access-token">{translate('dataset.access_token')}</a>
                <h1 className="af-title--content">{translate('dataset.active_datasets_counter')}{numberTotalItems}</h1>
                <div className="row row--datasets-filters">
                    <div className="col">
                        <div className="af-filter-inline">
                            <span className="af-filter-inline__title">
                                <span className="glyphicon glyphicon-filter"/>
                                <span className="af-filter-inline__title-text">{translate('dataset.filter_by')}</span>
                            </span>
                            <div className="af-filter-inline__field">
                                <label className="af-form__group-label" htmlFor="inputtext1">{translate('dataset.name')}</label>
                                <div className="af-form__text">
                                    <input className="af-form__input-text" id="inputtext1" name="inputtextname"
                                           onChange={onChangeFilter} type="text"/>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <EmptyArrayManager items={items} emptyArrayMessage={translate('dataset.empty_list')}>
                    <ItemsTable
                        items={items}
                        filters={filters}
                        onChangePaging={onChangePaging}
                        onChangeFilter={onChangeFilter}/>
                </EmptyArrayManager>
            </div>
        </>
    )
};

export default Home;
