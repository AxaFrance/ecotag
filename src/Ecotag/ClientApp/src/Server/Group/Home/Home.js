import React from 'react';
import Title from '../../../TitleBar';
import New from './New';

import './Home.scss';
import '../../shared/Modal/modal.scss';
import EmptyArrayManager from "../../../EmptyArrayManager";
import ItemsTable from "./ItemsTable";
import useProjectTranslation from "../../../useProjectTranslation";

const Home = ({
                  items,
                  numberItemsTotal,
                  filters,
                  onChangePaging,
                  fields,
                  isSubmitable,
                  hasSubmit,
                  onChangeCreateGroup,
                  onSubmitCreateGroup,
                  onUpdateUser,
              }) => {
    const {translate} = useProjectTranslation();

    return (
        <>
            <Title subtitle={translate('group.subtitle')} title={translate('group.title')}/>
            <div className="af-home container">
                <h1 className="af-title--content">{translate('group.creation_title')}</h1>
                <New
                    disabled={isSubmitable}
                    fields={fields}
                    hasSubmit={hasSubmit}
                    onChangeCreateGroup={onChangeCreateGroup}
                    onSubmitCreateGroup={onSubmitCreateGroup}
                />
                <h1 className="af-title--content">
                    {`${translate('group.count_teams.first_part')}${numberItemsTotal}${translate('group.count_teams.second_part')}`}
                </h1>
                <EmptyArrayManager items={items} emptyArrayMessage={translate('group.empty_list')}>
                    <ItemsTable
                        items={items}
                        filters={filters}
                        onChangePaging={onChangePaging}
                        onUpdateUser={onUpdateUser}
                    />
                </EmptyArrayManager>
            </div>
        </>
    );
};

export default Home;
