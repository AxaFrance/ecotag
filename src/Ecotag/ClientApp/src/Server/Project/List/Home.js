import React from 'react';
import {Link,} from 'react-router-dom';
import Title from '../../../TitleBar';
import './Home.scss';
import '../../shared/Modal/modal.scss';
import EmptyArrayManager from '../../../EmptyArrayManager';
import ItemsTable from './ItemsTable';
import {DataScientist} from '../../withAuthentication';
import useProjectTranslation from '../../../useProjectTranslation';

const Home = ({items, numberTotalItems, filters, onChangePaging, onChangeFilter, onChangeSort, fetch, user}) => {
    const {translate} = useProjectTranslation();

    return (
        <>
            <Title title={translate('project.projects_list.title')} subtitle={translate('project.projects_list.subtitle')}/>
            <div className="af-home container">
                {user.roles.includes(DataScientist) && <Link className="btn af-btn af-btn--quote" to="/projects/new">
                    <span className="af-btn__text">{translate('project.projects_list.new')}</span>
                </Link>}
                <h1 className="af-title--content">{`${translate('project.projects_list.active_projects_counter')} (${numberTotalItems})`}</h1>
                <div className="row row--projects-filters">
                    <div className="col">
                        <div className="af-filter-inline">
                            <span className="af-filter-inline__title">
                                <span className="glyphicon glyphicon-filter"/>
                                <span className="af-filter-inline__title-text">{translate('project.projects_list.filter_by')}</span>
                            </span>
                            <div className="af-filter-inline__field">
                                <label className="af-form__group-label" htmlFor="inputTextFilterProjects">
                                    {translate('project.projects_list.project_name')}
                                </label>
                                <div className="af-form__text">
                                    <input
                                        className="af-form__input-text"
                                        id="inputTextFilterProjects"
                                        name="inputTextFilterProjects"
                                        onChange={event => onChangeFilter(event.target.value)}
                                        type="text"
                                    />
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <EmptyArrayManager items={items} emptyArrayMessage={translate('project.projects_list.no_elements')}>
                    <ItemsTable
                        items={items}
                        filters={filters}
                        onChangePaging={onChangePaging}
                        onChangeSort={onChangeSort}
                        fetch={fetch}
                    />
                </EmptyArrayManager>
            </div>
        </>
    );
};

export default Home;
