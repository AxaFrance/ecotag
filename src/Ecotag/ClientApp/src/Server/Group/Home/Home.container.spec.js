import React from 'react';
import {createMemoryHistory} from "history";
import {BrowserRouter} from 'react-router-dom';
import {render, screen, waitFor} from '@testing-library/react';
import {HomeContainer} from './Home.container';
import sleep from "../../../sleep";
import {fireEvent} from "@testing-library/dom";

describe('Home.container for groups', () => {

    const history = createMemoryHistory({initialEntries: ['/']});
    const givenGroups = [{
        id: "0001",
        name: "developpeurs",
        userIds: ["0001", "0002"]
    }];
    const givenFetch = async (url) => {
        await sleep(1);
        switch (url) {
            case "groups":
                return {ok: true, json: () => Promise.resolve(givenGroups)};
            default:
                return {ok: true,
                    json: () => Promise.resolve([{id: "0001", email: "gilles.cruchon@axa.fr"}, {
                        id: "0002",
                        email: "guillaume.chervet@axa.fr"
                    }])
                };
        }
    };

    it('HomeContainer render correctly the groups', async () => {
        const {container, asFragment, getByText} = render(<BrowserRouter history={history}><HomeContainer
            fetch={givenFetch}/></BrowserRouter>);
        const messageEl = await waitFor(() => getByText('developpeurs'));
        expect(messageEl).toHaveTextContent(
            'developpeurs'
        );
        expect(asFragment()).toMatchSnapshot();

        const updateButton = container.querySelector('.glyphicon-pencil');
        fireEvent.click(updateButton);
        await waitFor(() => expect(getByText(/Supprimer/i)).not.toBeNull());

        const submitUpdateButton = screen.getByLabelText(/SubmitUpdate/i);
        fireEvent.click(submitUpdateButton);
        await waitFor(() => expect(screen.queryByText(/Supprimer/i)).toBeNull());
    });
});

