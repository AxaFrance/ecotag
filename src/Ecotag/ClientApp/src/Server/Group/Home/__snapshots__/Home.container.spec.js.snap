// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.container for groups Home.container translation should render groups correctly with english translation 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--backhome"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/"
        title="Home page"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Teams
      </h1>
      <small
        class="af-title-bar__subtitle"
      >
        Annotation teams governance
      </small>
    </div>
  </div>
  <div
    class="af-home container"
  >
    <h1
      class="af-title--content"
    >
      Creation
    </h1>
    <form
      name="newGroup"
    >
      <article
        class="af-panel af-panel--new-group"
      >
        <section
          class="af-panel__content af-panel__content--new-group"
        >
          <div
            class="row af-form__group"
          >
            <div
              class="col-md-2"
            >
              <label
                class="af-form__group-label"
                for="NAME"
              >
                Name : 
              </label>
            </div>
            <div
              class="col-md-10"
            >
              <div
                class="af-form__text"
              >
                <input
                  class="af-form__input-text"
                  id="NAME"
                  name="NAME"
                  type="text"
                  value=""
                />
              </div>
              <small
                class="af-form__help"
              >
                Ex : bank account details annotators
              </small>
            </div>
          </div>
          <button
            class="btn af-btn af-btn--hasiconRight af-btn--confirm af-btn--disabled"
            disabled=""
            id="myForm"
          >
            <span
              class="af-btn-text"
            >
              Validate
            </span>
            <i
              class="glyphicon glyphicon-arrowthin-right"
            />
          </button>
        </section>
      </article>
    </form>
    <h1
      class="af-title--content"
    >
      There is/are (1) team(s)
    </h1>
    <table
      class="af-table"
    >
      <thead
        class="af-table-thead"
      >
        <tr
          class="af-table__tr"
        >
          <th
            class="af-table__th"
          >
            <span
              class="af-table__th-content"
            >
              Name
            </span>
          </th>
          <th
            class="af-table__th"
          >
            <span
              class="af-table__th-content"
            >
              Users
            </span>
          </th>
          <th
            class="af-table__th"
          >
            <span
              class="af-table__th-content"
            >
              Actions
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        class="af-table__body"
      >
        <tr
          class="af-table__tr"
        >
          <td
            class="af-table__cell"
          >
            developers
          </td>
          <td
            class="af-table__cell"
          >
            gilles.cruchon@axa.fr, guillaume.chervet@axa.fr
          </td>
          <td
            class="af-table__cell af-table__small"
          >
            <a
              class="btn af-btn--circle"
              href="#"
              id="editActionId"
              role="button"
              tabindex="0"
              title="Edit"
            >
              <i
                class="glyphicon glyphicon-pencil"
              />
            </a>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="af-paging"
    >
      <div
        class="af-paging__limit"
      >
        <div
          class="af-paging__limit"
        >
          <form
            class="af-paging__form"
          >
            <div
              class="af-form__group"
            >
              <div
                class="col col-sm-2 col-md-2 col-lg-2 col-xl-2"
              >
                <label
                  class="af-form__group-label"
                  for="home_paging"
                >
                  Display
                </label>
              </div>
              <div
                class="col col-sm-10 col-md-10 col-lg-10 col-xl-10"
              >
                <div
                  class="af-form__select"
                >
                  <div
                    class="af-form__select-container"
                  >
                    <select
                      class="af-form__input-select"
                      id="home_paging"
                    >
                      <option
                        value="5"
                      >
                        5
                      </option>
                      <option
                        value="10"
                      >
                        10
                      </option>
                      <option
                        value="25"
                      >
                        25
                      </option>
                      <option
                        value="50"
                      >
                        50
                      </option>
                      <option
                        value="100"
                      >
                        100
                      </option>
                    </select>
                    <span
                      class="glyphicon glyphicon-menu-down"
                    />
                  </div>
                  <span
                    class="af-form__input-cmplt"
                  >
                    elements
                  </span>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
      <div
        class="af-paging__pager"
      >
        <nav
          class="af-pager"
        >
          <ul
            class="af-pager__pagination"
          >
            <li
              class="af-pager__item af-pager__item--disabled"
            >
              <span
                class="af-pager__item-nolink"
              >
                « Previous
              </span>
            </li>
            <li
              class="af-pager__item af-pager__item--active"
            >
              <a
                class="af-pager__item-link"
                href="#"
              >
                <span>
                  1
                </span>
              </a>
            </li>
            <li
              class="af-pager__item af-pager__item--disabled"
            >
              <span
                class="af-pager__item-nolink"
              >
                Next »
              </span>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home.container for groups HomeContainer render correctly the groups 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--backhome"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/"
        title="Home page"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Teams
      </h1>
      <small
        class="af-title-bar__subtitle"
      >
        Annotation teams governance
      </small>
    </div>
  </div>
  <div
    class="af-home container"
  >
    <h1
      class="af-title--content"
    >
      Creation
    </h1>
    <form
      name="newGroup"
    >
      <article
        class="af-panel af-panel--new-group"
      >
        <section
          class="af-panel__content af-panel__content--new-group"
        >
          <div
            class="row af-form__group"
          >
            <div
              class="col-md-2"
            >
              <label
                class="af-form__group-label"
                for="NAME"
              >
                Name : 
              </label>
            </div>
            <div
              class="col-md-10"
            >
              <div
                class="af-form__text"
              >
                <input
                  class="af-form__input-text"
                  id="NAME"
                  name="NAME"
                  type="text"
                  value=""
                />
              </div>
              <small
                class="af-form__help"
              >
                Ex : bank account details annotators
              </small>
            </div>
          </div>
          <button
            class="btn af-btn af-btn--hasiconRight af-btn--confirm af-btn--disabled"
            disabled=""
            id="myForm"
          >
            <span
              class="af-btn-text"
            >
              Validate
            </span>
            <i
              class="glyphicon glyphicon-arrowthin-right"
            />
          </button>
        </section>
      </article>
    </form>
    <h1
      class="af-title--content"
    >
      There is/are (1) team(s)
    </h1>
    <table
      class="af-table"
    >
      <thead
        class="af-table-thead"
      >
        <tr
          class="af-table__tr"
        >
          <th
            class="af-table__th"
          >
            <span
              class="af-table__th-content"
            >
              Name
            </span>
          </th>
          <th
            class="af-table__th"
          >
            <span
              class="af-table__th-content"
            >
              Users
            </span>
          </th>
          <th
            class="af-table__th"
          >
            <span
              class="af-table__th-content"
            >
              Actions
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        class="af-table__body"
      >
        <tr
          class="af-table__tr"
        >
          <td
            class="af-table__cell"
          >
            developers
          </td>
          <td
            class="af-table__cell"
          >
            gilles.cruchon@axa.fr, guillaume.chervet@axa.fr
          </td>
          <td
            class="af-table__cell af-table__small"
          >
            <a
              class="btn af-btn--circle"
              href="#"
              id="editActionId"
              role="button"
              tabindex="0"
              title="Edit"
            >
              <i
                class="glyphicon glyphicon-pencil"
              />
            </a>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="af-paging"
    >
      <div
        class="af-paging__limit"
      >
        <div
          class="af-paging__limit"
        >
          <form
            class="af-paging__form"
          >
            <div
              class="af-form__group"
            >
              <div
                class="col col-sm-2 col-md-2 col-lg-2 col-xl-2"
              >
                <label
                  class="af-form__group-label"
                  for="home_paging"
                >
                  Display
                </label>
              </div>
              <div
                class="col col-sm-10 col-md-10 col-lg-10 col-xl-10"
              >
                <div
                  class="af-form__select"
                >
                  <div
                    class="af-form__select-container"
                  >
                    <select
                      class="af-form__input-select"
                      id="home_paging"
                    >
                      <option
                        value="5"
                      >
                        5
                      </option>
                      <option
                        value="10"
                      >
                        10
                      </option>
                      <option
                        value="25"
                      >
                        25
                      </option>
                      <option
                        value="50"
                      >
                        50
                      </option>
                      <option
                        value="100"
                      >
                        100
                      </option>
                    </select>
                    <span
                      class="glyphicon glyphicon-menu-down"
                    />
                  </div>
                  <span
                    class="af-form__input-cmplt"
                  >
                    elements
                  </span>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
      <div
        class="af-paging__pager"
      >
        <nav
          class="af-pager"
        >
          <ul
            class="af-pager__pagination"
          >
            <li
              class="af-pager__item af-pager__item--disabled"
            >
              <span
                class="af-pager__item-nolink"
              >
                « Previous
              </span>
            </li>
            <li
              class="af-pager__item af-pager__item--active"
            >
              <a
                class="af-pager__item-link"
                href="#"
              >
                <span>
                  1
                </span>
              </a>
            </li>
            <li
              class="af-pager__item af-pager__item--disabled"
            >
              <span
                class="af-pager__item-nolink"
              >
                Next »
              </span>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home.container for groups should render groups correctly with french translation 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--backhome"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/"
        title="Home page"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Équipes
      </h1>
      <small
        class="af-title-bar__subtitle"
      >
        Gestion des équipes d'annotateurs
      </small>
    </div>
  </div>
  <div
    class="af-home container"
  >
    <h1
      class="af-title--content"
    >
      Création
    </h1>
    <form
      name="newGroup"
    >
      <article
        class="af-panel af-panel--new-group"
      >
        <section
          class="af-panel__content af-panel__content--new-group"
        >
          <div
            class="row af-form__group"
          >
            <div
              class="col-md-2"
            >
              <label
                class="af-form__group-label"
                for="NAME"
              >
                Nom : 
              </label>
            </div>
            <div
              class="col-md-10"
            >
              <div
                class="af-form__text"
              >
                <input
                  class="af-form__input-text"
                  id="NAME"
                  name="NAME"
                  type="text"
                  value=""
                />
              </div>
              <small
                class="af-form__help"
              >
                Ex : annotateurs de RIB
              </small>
            </div>
          </div>
          <button
            class="btn af-btn af-btn--hasiconRight af-btn--confirm af-btn--disabled"
            disabled=""
            id="myForm"
          >
            <span
              class="af-btn-text"
            >
              Valider
            </span>
            <i
              class="glyphicon glyphicon-arrowthin-right"
            />
          </button>
        </section>
      </article>
    </form>
    <h1
      class="af-title--content"
    >
      Il y a (1) équipe(s)
    </h1>
    <table
      class="af-table"
    >
      <thead
        class="af-table-thead"
      >
        <tr
          class="af-table__tr"
        >
          <th
            class="af-table__th"
          >
            <span
              class="af-table__th-content"
            >
              Nom
            </span>
          </th>
          <th
            class="af-table__th"
          >
            <span
              class="af-table__th-content"
            >
              Utilisateurs
            </span>
          </th>
          <th
            class="af-table__th"
          >
            <span
              class="af-table__th-content"
            >
              Actions
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        class="af-table__body"
      >
        <tr
          class="af-table__tr"
        >
          <td
            class="af-table__cell"
          >
            developers
          </td>
          <td
            class="af-table__cell"
          >
            gilles.cruchon@axa.fr, guillaume.chervet@axa.fr
          </td>
          <td
            class="af-table__cell af-table__small"
          >
            <a
              class="btn af-btn--circle"
              href="#"
              id="editActionId"
              role="button"
              tabindex="0"
              title="Modifier"
            >
              <i
                class="glyphicon glyphicon-pencil"
              />
            </a>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="af-paging"
    >
      <div
        class="af-paging__limit"
      >
        <div
          class="af-paging__limit"
        >
          <form
            class="af-paging__form"
          >
            <div
              class="af-form__group"
            >
              <div
                class="col col-sm-2 col-md-2 col-lg-2 col-xl-2"
              >
                <label
                  class="af-form__group-label"
                  for="home_paging"
                >
                  Afficher
                </label>
              </div>
              <div
                class="col col-sm-10 col-md-10 col-lg-10 col-xl-10"
              >
                <div
                  class="af-form__select"
                >
                  <div
                    class="af-form__select-container"
                  >
                    <select
                      class="af-form__input-select"
                      id="home_paging"
                    >
                      <option
                        value="5"
                      >
                        5
                      </option>
                      <option
                        value="10"
                      >
                        10
                      </option>
                      <option
                        value="25"
                      >
                        25
                      </option>
                      <option
                        value="50"
                      >
                        50
                      </option>
                      <option
                        value="100"
                      >
                        100
                      </option>
                    </select>
                    <span
                      class="glyphicon glyphicon-menu-down"
                    />
                  </div>
                  <span
                    class="af-form__input-cmplt"
                  >
                    éléments
                  </span>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
      <div
        class="af-paging__pager"
      >
        <nav
          class="af-pager"
        >
          <ul
            class="af-pager__pagination"
          >
            <li
              class="af-pager__item af-pager__item--disabled"
            >
              <span
                class="af-pager__item-nolink"
              >
                « Précédent
              </span>
            </li>
            <li
              class="af-pager__item af-pager__item--active"
            >
              <a
                class="af-pager__item-link"
                href="#"
              >
                <span>
                  1
                </span>
              </a>
            </li>
            <li
              class="af-pager__item af-pager__item--disabled"
            >
              <span
                class="af-pager__item-nolink"
              >
                Suivant »
              </span>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
